Framework/Interaction/Rest/Company.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2018 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Framework\Interaction\Rest;
+
+use ClassyLlama\AvaTax\Api\RestCompanyInterface;
+use ClassyLlama\AvaTax\Framework\Interaction\Rest;
+use Magento\Framework\DataObject;
+
+class Company extends Rest implements RestCompanyInterface
+{
+    /**
+     * @param \Avalara\AvaTaxClient $client
+     * @param DataObject|null       $request
+     *
+     * @return mixed
+     * @throws \ClassyLlama\AvaTax\Exception\AvataxConnectionException
+     */
+    protected function getCompaniesFromClient( $client, $request = null )
+    {
+        if ($request === null)
+        {
+            $request = $this->dataObjectFactory->create();
+        }
+
+        $clientResult = $client->queryCompanies(
+            $request->getInclude(),
+            $request->getFilter(),
+            $request->getTop(),
+            $request->getSkip(),
+            $request->getOrderBy()
+        );
+
+        $this->validateResult( $clientResult, $request );
+
+        return $this->formatResult( $clientResult )->getValue();
```

Comment:
@LordZardeck - I don't understand the getValue call here.  formatResult returns either an array or a DataObject (which doesn't have a getValue method).

Reply:
Ahh, this is why I dislike returning generic DataObjects, and always prefer returning interfaces. `value` is a data property on the DataObject, so I'm calling `getValue` to retrieve that. I would have caught this if I had proper return types instead of just `mixed` as PHPStorm complains about magic method calls. Updating code to be more clear on this matter